hrms/hr/doctype/leave_application/leave_application.py

```diff
@@ -350,6 +350,8 @@ def validate_block_days(self):
 			frappe.throw(_("You are not authorized to approve leaves on Block Dates"), LeaveDayBlockedError)
 
 	def validate_balance_leaves(self):
+		precision = cint(frappe.db.get_single_value("System Settings", "float_precision", cache=True))
```

Comment:
```suggestion
		precision = cint(frappe.db.get_single_value("System Settings", "float_precision")) || 2
```
- Added a fallback precision if nothing is set in System Settings. Also set this fallback precision for the functions displaying the leave balance
- removed cache flag since its True by default in the framework now
<img width="722" alt="image" src="https://github.com/user-attachments/assets/882acdc9-a505-4b77-939b-7aa9584777c5">